tests/aldn1000.py

```diff
@@ -0,0 +1,199 @@
+    def test_GIVEN_new_diameter_WHEN_set_diameter_THEN_new_diameter_set(self, _, value):
+        expected_diameter = value
+        self.ca.set_pv_value("DIAMETER:SP", expected_diameter)
+
+        self.ca.assert_that_pv_is("DIAMETER", expected_diameter, timeout=2)
+
+    @parameterized.expand([('Value 1', 12345), ('Value 2', 1234), ('Value 3', 77424)])
+    @skip_if_recsim("Not supported in RECSIM")
+    def test_GIVEN_new_invalid_high_diameter_WHEN_set_diameter_THEN_diameter_set_limit_returned(self, _, value):
+        invalid_diameter = value
+        expected_diameter = 1000.00
+        self.ca.set_pv_value("DIAMETER:SP", invalid_diameter)
+
+        self.ca.assert_that_pv_is("DIAMETER", expected_diameter, timeout=2)
+
+    @parameterized.expand([('Value 1', -2345), ('Value 2', -1234), ('Value 3', -0676424)])
+    @skip_if_recsim("Not supported in RECSIM")
+    def test_GIVEN_new_invalid_low_diameter_WHEN_set_diameter_THEN_diameter_set_limit_returned(self, _, value):
+        invalid_diameter = value
+        expected_diameter = 0.00
+        self.ca.set_pv_value("DIAMETER:SP", invalid_diameter)
+
+        self.ca.assert_that_pv_is("DIAMETER", expected_diameter, timeout=2)
+
+    @parameterized.expand([('Value 1', 14.1), ('Value 2', 24.23), ('Value 3', 30)])
+    @skip_if_recsim("Not supported in RECSIM")
+    def test_GIVEN_new_diameter_above_14mm_WHEN_set_diameter_THEN_volume_units_changed(self, _, value):
+        set_diameter = value
+        expected_units = 'mL'
+        self.ca.set_pv_value("DIAMETER:SP", set_diameter)
+
+        self.ca.assert_that_pv_is("VOLUME:UNITS", expected_units, timeout=2)
+
+    @parameterized.expand([('Value 1', 14.0), ('Value 2', 10.0), ('Value 3', 5.0)])
+    @skip_if_recsim("Not supported in RECSIM")
+    def test_GIVEN_new_diameter_below_or_eq_14mm_WHEN_set_diameter_THEN_volume_units_changed(self, _, value):
+        set_diameter = value
+        expected_units = 'uL'
+        self.ca.set_pv_value("DIAMETER:SP", set_diameter)
+
+        self.ca.assert_that_pv_is("VOLUME:UNITS", expected_units, timeout=2)
+
+    @parameterized.expand([('Value 1', 0.123), ('Value 2', 1.342), ('Value 3', 12.34)])
+    @skip_if_recsim("Not supported in RECSIM")
+    def test_GIVEN_new_volume_WHEN_set_volume_THEN_new_volume_set(self, _, value):
+        expected_volume = value
+        self.ca.set_pv_value("VOLUME:SP", expected_volume)
+
+        self.ca.assert_that_pv_is("VOLUME", expected_volume, timeout=2)
+
+    @parameterized.expand([('Direction 1', 'Withdraw'), ('Direction 2', 'Infuse')])
+    def test_GIVEN_new_direction_WHEN_set_direction_THEN_new_direction_set(self, _, direction):
+        expected_direction = direction
+        self.ca.set_pv_value("DIRECTION:SP", expected_direction)
+
+        self.ca.assert_that_pv_is("DIRECTION", expected_direction)
+
+    @skip_if_recsim("Not supported in RECSIM")
+    def test_GIVEN_existing_direction_WHEN_set_reverse_direction_THEN_direction_reversed(self):
```

Comment:
This is only testing one direction (whichever is the default on the emulator) can you make it test both.

Reply:
Created new parameterized test that checks multiple cases for flow direction.